src/test/ram_statement_equal_clone_test.cpp

```diff
@@ -0,0 +1,470 @@
+    delete c;
+}
+
+TEST(RamClear, CloneAndEquals) {
+    // CLEAR A
+    RamRelation A("A", 1, 1, {"x"}, {"i"}, RelationRepresentation::DEFAULT);
+    RamClear a(std::make_unique<RamRelationReference>(&A));
+    RamClear b(std::make_unique<RamRelationReference>(&A));
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamClear* c = a.clone();
+    EXPECT_EQ(a, *c);
+    EXPECT_NE(&a, c);
+    delete c;
+}
+
+TEST(RamMerge, CloneAndEquals) {
+    // MERGE B WITH A
+    RamRelation A("A", 1, 1, {"x"}, {"i"}, RelationRepresentation::DEFAULT);
+    RamRelation B("B", 1, 1, {"x"}, {"i"}, RelationRepresentation::DEFAULT);
+    RamMerge a(std::make_unique<RamRelationReference>(&B), std::make_unique<RamRelationReference>(&A));
+    RamMerge b(std::make_unique<RamRelationReference>(&B), std::make_unique<RamRelationReference>(&A));
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamMerge* c = a.clone();
+    EXPECT_EQ(a, *c);
+    EXPECT_NE(&a, c);
+    delete c;
+}
+
+TEST(RamSwap, CloneAndEquals) {
+    // SWAP(A,B)
+    RamRelation A("A", 1, 1, {"x"}, {"i"}, RelationRepresentation::DEFAULT);
+    RamRelation B("B", 1, 1, {"x"}, {"i"}, RelationRepresentation::DEFAULT);
+    RamSwap a(std::make_unique<RamRelationReference>(&A), std::make_unique<RamRelationReference>(&B));
+    RamSwap b(std::make_unique<RamRelationReference>(&A), std::make_unique<RamRelationReference>(&B));
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamSwap* c = a.clone();
+    EXPECT_EQ(a, *c);
+    EXPECT_NE(&a, c);
+    delete c;
+}
+
+TEST(RamQuery, CloneAndEquals) {
+    RamRelation A("A", 3, 1, {"a", "b", "c"}, {"i", "s", "i"}, RelationRepresentation::DEFAULT);
+    RamRelation B("B", 2, 1, {"a", "c"}, {"i", "i"}, RelationRepresentation::DEFAULT);
+    /*
+     * QUERY
+     *  FOR t0 IN A
+     *   PROJECT (t0.0, t0.2) INTO B
+     */
+    std::vector<std::unique_ptr<RamExpression>> a_expressions;
+    a_expressions.emplace_back(new RamTupleElement(0, 0));
+    a_expressions.emplace_back(new RamTupleElement(0, 2));
+    auto a_project = std::make_unique<RamProject>(
+            std::make_unique<RamRelationReference>(&B), std::move(a_expressions));
+    auto a_scan = std::make_unique<RamScan>(
+            std::make_unique<RamRelationReference>(&A), 0, std::move(a_project), "");
+
+    std::vector<std::unique_ptr<RamExpression>> b_expressions;
+    b_expressions.emplace_back(new RamTupleElement(0, 0));
+    b_expressions.emplace_back(new RamTupleElement(0, 2));
+    auto b_project = std::make_unique<RamProject>(
+            std::make_unique<RamRelationReference>(&B), std::move(b_expressions));
+    auto b_scan = std::make_unique<RamScan>(
+            std::make_unique<RamRelationReference>(&A), 0, std::move(b_project), "");
+
+    RamQuery a(std::move(a_scan));
+    RamQuery b(std::move(b_scan));
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamQuery* c = a.clone();
+    EXPECT_EQ(a, *c);
+    EXPECT_NE(&a, c);
+    delete c;
+
+    /*
```

Comment:
Split this up into multiple tests of RamQuery. More generally, it's fine to have multiple tests of the same thing, if they test different aspects.